### PR TITLE
Relax the gast version requirement

### PR DESCRIPTION
**Pull request by klecki**
## Description:
Remove the limit on gast version.
Handle the missing type_comment parameter in gast.Assign usages.

### Affected modules and functionalities:
Gast, AG

### Tests:

- [x] Existing tests apply
Conditionals/auto_aug should work as previously.
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: dali/python/nvidia/dali/_autograph/pyct/testing/codegen.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def generate_Assign(self):
     # Generate right-hand side
     value_node = self.generate_expression()
     # Put it all together
-    node = gast.Assign(targets=[target_node], value=value_node)
+    node = gast.Assign(targets=[target_node], value=value_node, type_comment=None)
```
Comment (JanuszL) [inline dali/python/nvidia/dali/_autograph/pyct/testing/codegen.py:170]: Would it work with older gast?

Comment (klecki) [reply dali/python/nvidia/dali/_autograph/pyct/testing/codegen.py:170]: And without this change on an older gast we get
```
======================================================================
ERROR: test_replace_code_block (pyct.test_templates.TemplatesTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/klecki/work/salvador/dali/dali/test/python/autograph/pyct/test_templates.py", line 114, in test_replace_code_block
    gast.Assign(
  File "/home/klecki/.local/lib/python3.10/site-packages/gast/gast.py", line 26, in create_node
    raise TypeError(
TypeError: Assign constructor takes either 0 or 3 mandatory arguments

```

Comment (JanuszL) [reply dali/python/nvidia/dali/_autograph/pyct/testing/codegen.py:170]: 😱

**Comment by thomas-bouvier** [comment]
@klecki, I wrote some changes starting from the version in this PR. Where should I submit my PR?

**Comment by JanuszL** [comment]
> @klecki, I wrote some changes starting from the version in this PR. Where should I submit my PR?

Hi @thomas-bouvier,

Please fork the DALI repo, create a branch with your changes there, and then create a PR to the main repository. You can find more info on the [GitHub docs page](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).

**Comment by klecki** [comment]
Improved version submitted as a separate PR.

**Pull request closed without merging**